src/views/FarmAuction/helpers.tsx

```diff
@@ -1,3 +1,166 @@
-// eslint-disable-next-line import/prefer-default-export
+import { toDate, add, hoursToSeconds, intervalToDuration } from 'date-fns'
+import { BSC_BLOCK_TIME, DEFAULT_TOKEN_DECIMAL } from 'config'
+import { getBidderInfo } from 'config/constants/farmAuctions'
+import { simpleRpcProvider } from 'utils/providers'
+import { AuctionsResponse, FarmAuctionContractStatus, BidsPerAuction, ViewBidderAuctionsResponse } from 'utils/types'
+import { Auction, AuctionStatus, Bidder, BidderAuction } from 'config/constants/types'
+import { ethersToBigNumber } from 'utils/bigNumber'
+
 export const FORM_ADDRESS =
   'https://docs.google.com/forms/d/e/1FAIpQLScUkwbsMWwg7L5jjGjEcmv6RsoCNhFDkV3xEpRu2KcJrr47Sw/viewform'
+
+// Sorts bidders received from smart contract by bid amount in descending order (biggest -> smallest)
+// Also ammends bidder information with getBidderInfo
+// auction is required if data will be used for table display, hence in reclaim and congratulations card its ommited
+export const sortAuctionBidders = (bidders: BidsPerAuction[], auction?: Auction): Bidder[] => {
+  const sortedBidders = [...bidders]
+    .sort((a, b) => {
+      if (a.amount.lt(b.amount)) {
+        return 1
+      }
+      if (a.amount.gt(b.amount)) {
+        return -1
+      }
+      return 0
+    })
+    .map((bidder, index) => {
+      const bidderInfo = getBidderInfo(bidder.account)
+      return {
+        ...bidderInfo,
+        position: index + 1,
+        account: bidder.account,
+        amount: bidder.amount,
+      }
+    })
+
+  // Positions need to be adjusted in case 2 bidders has the same bid amount
+  let adjustedPosition = 1
+
+  return sortedBidders.map((bidder, index, unadjustedBidders) => {
+    const amount = ethersToBigNumber(bidder.amount)
+    if (index !== 0) {
+      const samePositionAsAbove = bidder.amount.eq(unadjustedBidders[index - 1].amount)
+      adjustedPosition = samePositionAsAbove ? adjustedPosition : adjustedPosition + 1
+      // Reclaim and congratulations card don't need auction data or isTopPosition
+      // in this case it is set to false just to avoid TS errors
+      let isTopPosition = auction ? adjustedPosition <= auction.topLeaderboard : false
+      // This is here in case we closed auction with less/more winners for some reason
+      if (auction && auction.leaderboardThreshold.gt(0)) {
+        isTopPosition = auction.leaderboardThreshold.lte(amount)
+      }
+      return {
+        ...bidder,
+        position: adjustedPosition,
+        isTopPosition,
+        samePositionAsAbove,
+        amount,
+      }
+    }
+    return {
+      ...bidder,
+      isTopPosition: true,
+      samePositionAsAbove: false,
+      amount,
+    }
+  })
+}
+
+// Determine if the auction is:
+// - Live and biddable
+// - Has been scheduled for specific future date
+// - Not annoucned yet
+// - Recently Finished/Closed
+const getAuctionStatus = (
```

Comment:
ToBeAnnounced = nothing scheduled yet, UI will show "to be announced" and no timer. This is primarily used for the very first auction, after that "Closed" will be similar to ToBeAnnounced
Pending = auction was schedule but not started yet, UI will show dates and timer will be counting down to start date
Open = auction has started, UI shows leaderboard, bid button, timer counts down to end date
Finished = auction endblock has been reached, but SC didn't officially close the auction yet, UI hides bid button, timer says "Closing..."
Closed = SC closed the auction, leaderboard will show latest auction leaderboard, other UI elements are similar to ToBeAnounced (dates show "to be announced", timer is hidden, etc.)